torch_xla/csrc/tensor_util.cpp

```diff
@@ -671,9 +662,16 @@ at::Tensor MakeTensorFromXlaLiteral(const xla::Literal& literal,
 }
 
 bool TensorCompare(const at::Tensor& t1, const at::Tensor& t2) {
-  // Use TensorMemoryCompare until pytorch equal resolve the issue of
-  // non-deterministic ATEN compare of NAN.
-  return TensorMemoryCompare(t1, t2);
+  if (t1.scalar_type() != t2.scalar_type() || t1.sizes() != t2.sizes()) {
+    return false;
+  }
+  // PyTorch currently has an issue comparing tensors which have NaN values in
```

Comment:
Btw it's hard to get it "fixed" on pytorch side since comparing to Nan is defined to always return False. https://en.wikipedia.org/wiki/NaN#Comparison_with_NaN
This is also true for Numpy. :(

Reply:
The issue here was 😄 
```
T0: (t1 == t2) == true
T1: (t1 == t2) == false
```